### DISCUSSION TITLE
Release downloads that redirect to objects.githubusercontent.com failing due to expired/wrong certificate

### DISCUSSION THREAD

**Discussion by brokenjacobs** [community/community#50963] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

As an example:
`https://github.com/elsesiy/kubectl-view-secret/releases/download/v0.10.1/kubectl-view-secret_v0.10.1_darwin_arm64.tar.gz`

Redirects to what looks like an AWS bucket presented from objects.githubusercontent.com which presents an expired certificate for *.github.io:
<img width="672" alt="image" src="https://user-images.githubusercontent.com/3159463/227637786-4b92609c-b616-4e32-ae45-202ebd7828f6.png">

Something is broken it looks like, not sure how widespread this is.

**Comment by brokenjacobs** [comment]
What's funny is this appears to affect user-images.githubusercontent.com as well because the screenshot I uploaded won't show in my original post... 
In case it's helpful since the screenshot won't load:
```
Common Name (CN)	*.github.io
Organization (O)	GitHub, Inc.
Organizational Unit (OU)	<Not Part Of Certificate>
Common Name (CN)	DigiCert TLS RSA SHA256 2020 CA1
Organization (O)	DigiCert Inc
Organizational Unit (OU)	<Not Part Of Certificate>
Issued On	Thursday, March 17, 2022 at 6:00:00 PM
Expires On	Tuesday, March 21, 2023 at 5:59:59 PM
SHA-256 Fingerprint	7B 18 D8 BB A5 D1 0B 0D 50 AB E1 6B 45 1F A1 8F
8A FC 0C 07 13 63 68 43 0D FB 64 33 42 1F 39 FA
SHA-1 Fingerprint	8F 0E 79 24 71 C5 A7 D2 A7 46 76 30 C1 3C B7 2A
13 B0 01 B2

**Comment by PenelopeFudd** [comment]
```
$ curl -L  'https://github.com/krallin/tini/releases/download/v0.19.0/tini'
(Start: 2023-03-24 13:54:48)
curl: (60) SSL certificate problem: certificate has expired
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

**Comment by kglueck** [comment]
I can confirm this from my perspective too.

```
% openssl s_client -servername objects.githubusercontent.com -connect objects.githubusercontent.com:443 2>/dev/null | openssl x509 -noout -dates
notBefore=Mar 18 00:00:00 2022 GMT
notAfter=Mar 21 23:59:59 2023 GMT
```

**Comment by rekkih** [comment]
Happening pretty much the same as above for these for me as well:
- avatars.githubusercontent.com
- raw.githubusercontent.com

Thought it was just me locally, but guess not?

**Reply by rekkih** [reply]
Annnnd we're back?

**Reply by matthenry87** [reply]
Yup, icons are now loading on GitHub itself again.

**Comment by matthenry87** [comment]
Also experiencing this:

```bash
❯ curl -L https://github.com/bell-sw/Liberica/releases/download/17.0.6+10/bellsoft-jre17.0.6+10-linux-amd64.tar.gz
curl: (60) SSL certificate problem: certificate has expired
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

**Comment by kglueck** [comment]
Yup, we're good again too.  =)

**Comment by Nuxij** [comment]
This is still happening to me in a docker container

**Reply by rewathkafley** [reply]
It just started happening to me few moments ago while trying to install docker compose

<img width="1031" alt="image" src="https://github.com/community/community/assets/9987781/6b937d48-4bbf-4368-a88f-3a5b9bd2f048">

**Reply by juniorz** [reply]
Example of the problem

```
$ docker run -it ubuntu:latest
root@155101d31680:/# cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
root@155101d31680:/# apt-get update -q && apt-get install -qy curl
root@155101d31680:/# curl -sSL "https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv5.1.0/kustomize_v5.1.0_linux_amd64.tar.gz"
curl: (60) SSL: no alternative certificate subject name matches target host name 'objects.githubusercontent.com'
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

(some) certificate has the SAN though:

```
root@155101d31680:/# echo "Q" | openssl s_client -connect objects.githubusercontent.com:443 -servername objects.githubusercontent.com -showcerts | openssl x509 -text -noout

[...]
            X509v3 Subject Alternative Name:
                DNS:*.github.io, DNS:github.io, DNS:*.github.com, DNS:github.com, DNS:www.github.com, DNS:*.githubusercontent.com, DNS:githubusercontent.com
[...]
```

**Reply by sibidharan** [reply]
```
root@essentials:/var/labsdata# wget -qO- https://code-server.dev/install.sh | sh
Ubuntu 22.04.2 LTS
Installing v4.14.1 of the amd64 deb package from GitHub.

+ mkdir -p ~/.cache/code-server
+ curl -#fL -o ~/.cache/code-server/code-server_4.14.1_amd64.deb.incomplete -C - https://github.com/coder/code-server/releases/download/v4.14.1/code-server_4.14.1_amd64.deb
curl: (60) SSL: no alternative certificate subject name matches target host name 'objects.githubusercontent.com'
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

Same is happening again while trying to install code-server

**Reply by juniorz** [reply]
There's likely no telemetry for this on Github side since the error happens at the clients, so hopefully someone with friends at Github spread the word.

**Reply by rewathkafley** [reply]
It seems we all have one more thing in common, OS is Linux/Ubuntu 22.04.

**Reply by juniorz** [reply]
Yep. It works on `ubuntu:23.04`

**Reply by juniorz** [reply]
Interestingly: https://discourse.ubuntu.com/t/jammy-jellyfish-release-notes/24668

> We’ve upgraded the OpenSSL library to the new 3.0 version, which disables a lot of legacy algorithms by default, as detailed in their [migration guide](https://www.openssl.org/docs/manmaster/man7/migration_guide.html). In particular, certificates using SHA1 or MD5 as hash algorithms are now invalid under the default security level.

**Reply by moorej-oci** [reply]
+1 happens on 22.04 but not 23.04

```
docker run --rm -it ubuntu:latest bash
# once loaded:
apt-get update -q && apt-get install -qy curl
# curl any example URL
curl -L https://github.com/bell-sw/Liberica/releases/download/17.0.6+10/bellsoft-jre17.0.6+10-linux-amd64.tar.gz --output file.tar.gz
# failure
# curl: (60) SSL: no alternative certificate subject name matches target host name 'objects.githubusercontent.com'
# More details here: https://curl.se/docs/sslcerts.html
# 
# curl failed to verify the legitimacy of the server and therefore could not
# establish a secure connection to it. To learn more about this situation and
# how to fix it, please visit the web page mentioned above.
```

```
docker run --rm -it ubuntu:23.04 bash
# once loaded:
apt-get update -q && apt-get install -qy curl
# curl any example URL
curl -L https://github.com/bell-sw/Liberica/releases/download/17.0.6+10/bellsoft-jre17.0.6+10-linux-amd64.tar.gz --output file.tar.gz
# success!
```

**Reply by PenelopeFudd** [reply]
On the server side, if they see clients connect, receive the server certificate, and disconnect without sending any data, that's the sign that the clients don't trust the server's certificate.  Possible reasons:
* The certificate has expired - easy to check
* One of the certificates in the chain of certificates has expired or been revoked - checkable
* The certificate is not signed by a CA that the client trusts - have to be a mind reader

That'd be the telemetry that they'd have to look at.

**Reply by daniel-san-pedro** [reply]
Changing to wget has allowed me to continue building my docker container for now.

**Reply by beelesnik** [reply]
Have same issue today while buidling docker image. As part of image build there is step to download https://github.com/tfsec/tfsec/releases/download/v1.28.0/tfsec-linux-amd64 and https://github.com/terraform-docs/terraform-docs/releases/download/v0.16.0/terraform-docs-v0.16.0-linux-amd64.tar.gz 
So, I got
`curl: (60) SSL: no alternative certificate subject name matches target host name 'objects.githubusercontent.com'`
